--- beta/usr/lib/hwsupport/sdcard-mount.sh	2022-05-09 23:18:38.000000000 +0200
+++ work/usr/lib/hwsupport/sdcard-mount.sh	2022-05-12 16:03:44.110844712 +0200
@@ -67,9 +67,18 @@
     #    OPTS+=",users,gid=100,umask=000,shortname=mixed,utf8=1,flush"
     #fi
 
-    # We need symlinks for Steam for now, so only automount ext4 as that'll Steam will format right now
-    if [[ ${ID_FS_TYPE} != "ext4" ]]; then
-      exit 1
+    if [[ -f /etc/default/steamos-btrfs ]]; then
+        source /etc/default/steamos-btrfs
+    fi
+    if [[ ${ID_FS_TYPE} == "ext4" ]]; then
+        OPTS="${STEAMOS_BTRFS_SDCARD_EXT4_MOUNT_OPTS:-rw,noatime,lazytime}"
+    elif [[ ${ID_FS_TYPE} == "f2fs" ]]; then
+        OPTS="${STEAMOS_BTRFS_SDCARD_F2FS_MOUNT_OPTS:-rw,noatime,lazytime,compress_algorithm=zstd,compress_chksum,whint_mode=fs-based,atgc,gc_merge}"
+    elif [[ ${ID_FS_TYPE} == "btrfs" ]]; then
+        OPTS="${STEAMOS_BTRFS_SDCARD_BTRFS_MOUNT_OPTS:-rw,noatime,lazytime,compress-force=zstd,space_cache=v2,autodefrag}"
+    else
+        # We need symlinks for Steam for now
+        exit 1
     fi
 
     if ! /bin/mount -o ${OPTS} ${DEVICE} ${MOUNT_POINT}; then
@@ -78,6 +87,17 @@
         exit 1
     fi
 
+    # Check if there's a subvolume @ and use that as root of the drive
+    if [[ ${ID_FS_TYPE} == "btrfs" ]] && btrfs subvolume show "${MOUNT_POINT}/${STEAMOS_BTRFS_SDCARD_BTRFS_MOUNT_SUBVOL:-@}" &>/dev/null; then
+        /bin/umount -f -l ${MOUNT_POINT}
+        OPTS+=",subvol=${STEAMOS_BTRFS_SDCARD_BTRFS_MOUNT_SUBVOL:-@}"
+        if ! /bin/mount -o ${OPTS} ${DEVICE} ${MOUNT_POINT}; then
+            echo "Error mounting ${DEVICE} (status = $?)"
+            /bin/rmdir ${MOUNT_POINT}
+            exit 1
+        fi
+    fi
+
     chown 1000:1000 ${MOUNT_POINT}
 
     echo "**** Mounted ${DEVICE} at ${MOUNT_POINT} ****"
