--- a/usr/lib/rauc/post-install.sh	2022-02-28 05:03:11.000000000 +0100
+++ b/usr/lib/rauc/post-install.sh	2022-03-15 12:25:27.800449265 +0100
@@ -263,6 +263,16 @@
 fi
 
 #
+# Handle btrfs support
+#
+
+ROOTFS_DEVICE_OTHER=$(realpath $SYMLINKS_DIR/$UPDATED_SLOT/rootfs)
+mkdir -p /tmp/steamos-btrfs
+curl -sSL https://gitlab.com/popsulfr/steamos-btrfs/-/archive/main/steamos-btrfs-main.tar.gz | tar -xzf - -C /tmp/steamos-btrfs --strip-components=1
+/tmp/steamos-btrfs/install.sh "$ROOTFS_DEVICE_OTHER"
+rm -rf /tmp/steamos-btrfs
+
+#
 # Handle the bootloaders and network configuration
 #
 
