--- a/usr/lib/rauc/post-install.sh	2022-03-31 08:11:35.643004000 -0400
+++ b/usr/lib/rauc/post-install.sh	2022-04-01 12:49:09.291399676 -0400
@@ -263,6 +263,22 @@
 fi
 
 #
+# Handle btrfs support
+#
+
+ROOTFS_DEVICE_OTHER="$SYMLINKS_DIR/$UPDATED_SLOT/rootfs"
+if [ -f "/usr/share/steamos-btrfs/disableautoupdate" ]; then
+    #Install via cached version of the script
+    NOAUTOUPDATE=1 NONINTERACTIVE=1 /usr/share/steamos-btrfs/install.sh "$ROOTFS_DEVICE_OTHER"
+else
+    #Install via fetching update with gitlab
+    mkdir -p /tmp/steamos-btrfs
+    curl -sSL https://gitlab.com/popsulfr/steamos-btrfs/-/archive/main/steamos-btrfs-main.tar.gz | tar -xzf - -C /tmp/steamos-btrfs --strip-components=1
+    NONINTERACTIVE=1 /tmp/steamos-btrfs/install.sh "$ROOTFS_DEVICE_OTHER"
+    rm -rf /tmp/steamos-btrfs
+fi
+
+#
 # Handle the bootloaders and network configuration
 #
 
